FrontEnd/src/components/admin/SubjectAdministration.js

```diff
@@ -1,12 +1,232 @@
+
+    const handleCreateSubjectClick = () => {
+        setShowNewSubjectInputField(true);
+    }
+
+    let subjectEditTextfield = "";
+    const handleEditSubjectClick = (_editedSubjectId, editedSubjectName) => {
+        setEditedSubjectId(_editedSubjectId);
+        subjectEditTextfield = editedSubjectName;
+    }
+
+    //Neues Fach erstellen
+    let newSubjectNameField = "";
+    const handleSaveSubject = () => {
+        if(newSubjectNameField === "") {
+            setNewSubjectInputFieldErrorMessage("Der Name des Fachs darf nicht leer sein");
+            return;
+        }
+
+        const setNewSubjectIdAndSetNewSubjectList = (newSubjectId) => {
+            const newSubjectList = subjectList;
+            newSubjectList.unshift({subjectId: newSubjectId.subjectId, subjectName: newSubjectNameField, deactivated: false});
+            setSubjectList(newSubjectList);
+            setNewSubjectInputFieldErrorMessage("");
+            setShowNewSubjectInputField(false);
+            newSubjectNameField = "";
+        }
+        const requestBody = JSON.stringify({
+            request_token: props.request_token,
+            name: newSubjectNameField
+        })
+        fetch(SERVER + "/api/subject/create",
+            {
+                "method": "POST",
+                "headers": {'Content-Type': 'application/json'},
+                "body": requestBody,
+            }).then(response => response.json())
+            .then((data) => (setNewSubjectIdAndSetNewSubjectList(data)))
+    }
+
+    //Fach editieren
+    const handleSaveEditedSubjectName = () => {
+        for (let i = 0; i< subjectList.length; i++){
+            if(subjectList[i].subjectId === editedSubjectId){
+                subjectList[i].subjectName = subjectEditTextfield;
+                const requestBody = JSON.stringify({subjectId: editedSubjectId, newName: subjectEditTextfield,request_token: props.request_token})
+                fetch(SERVER + "/api/subject/edit",
+                    {
+                        "method": "POST",
+                        "headers": {'Content-Type': 'application/json'},
+                        "body": requestBody
+                    })
+                setEditedSubjectId(null);
+                break;
+            }
+        }
+    }
+
+    //Fach aktivieren/deaktivieren
+    const changeSubjectStatus = (subjectId, deactivate) => {
+
+        const requestBody = JSON.stringify(
+            {
+                subjectId: subjectId ,
+                deactivated: deactivate,
+                request_token: props.request_token,
+            })
+        fetch(SERVER + "/api/subject/setDeactivated",
+            {
+                "method": "POST",
+                "headers": {'Content-Type': 'application/json'},
+                "body": requestBody
+            });
+
+        for (let i = 0; i < subjectList.length; i++){
+            if(subjectList[i].subjectId === subjectId){
+                const newSubjectList = subjectList;
```

Comment:
wird hier, wenn du veränderungen in newSubjectList vornimmst nicht auch subjectList bearbeitet? Ich glaube bei der deklaration zeigen beide variablen auf den gleichen Speicher

Reply:
Hast recht. Gut gesehen!